json-endpoints/jetpack/class.jetpack-json-api-plugins-modify-endpoint.php

```diff
@@ -117,19 +129,43 @@ protected function activate() {
 			}
 			$this->log[ $plugin ][] = __( 'Plugin activated.', 'jetpack' );
 		}
-		if ( ! $this->bulk && isset( $has_errors ) ) {
+
+		if ( ! $this->bulk && isset( $has_errors )  ) {
 			$plugin = $this->plugins[0];
+			if ( $permission_error ) {
+				return new WP_Error( 'unauthorized_error', $this->log[ $plugin ]['error'], 404 );
+			}
 			return new WP_Error( 'activation_error', $this->log[ $plugin ]['error'] );
 		}
 	}
 
+	protected function current_user_can( $capability, $plugin = null ) {
+		global $wp_version;
+		if ( version_compare( $wp_version, '4.9-beta2' ) >= 0 ) {
+			if ( $plugin ) {
+				return current_user_can( $capability, $plugin );
+			}
+			return current_user_can( $capability );
+		}
+		// Assume that the user has the capability...
```

Comment:
Should we err on the safe side of things and instead fallback to not allowing things?

Reply:
We are checking for other permissions already. This this checks are only for future version of jetpack. I think we will be able to remove the version check when we don't stop supporting older versions of WP.